plugins/urls/url.py

```diff
@@ -0,0 +1,43 @@
+import idna
+from kitchen.text.converters import to_unicode
+
+__author__ = 'Gareth Coles'
+
+
+class URL(object):
+    plugin = None
+
+    protocol = None
+    auth = None
+    domain = None
+    port = None
+    path = None
+
+    @property
+    def text(self):
```

Comment:
Indeed, repr and str each have their uses (and it might be wise to implement `__repr__` too).